pkg/agent/clientset.go

```diff
@@ -63,6 +64,8 @@ type ClientSet struct {
 	warnOnChannelLimit bool
 
 	syncForever bool // Continue syncing (support dynamic server count).
+
+	respectReceivedServerCount bool // Respect server count received from proxy server rather than relying on the agent's own server counter
```

Comment:
Should chat on this. During certain events with multiple KASs/Konnectivity servers we can get different server counts from different servers.

Reply:
Yeah, @avrittrohwer also raised this issue. One potential solution would be to cache the last n received server counts and take the highest value. Ultimately this gets solved by having proxy agents count the leases themselves.